core/src/main/java/nl/nn/adapterframework/jms/JmsListenerBase.java

```diff
@@ -134,49 +134,43 @@ public void close() {
 	 * @param rawMessage - Original message received, can not be <code>null</code>
 	 * @param threadContext - Thread context to be populated, can not be <code>null</code>
 	 */
-	public String getIdFromRawMessage(javax.jms.Message rawMessage, Map<String, Object> threadContext) throws ListenerException {
-		TextMessage message = null;
-		try {
-			message = (TextMessage) rawMessage;
-		} catch (ClassCastException e) {
-			log.error("message received by listener on ["+ getDestinationName()+ "] was not of type TextMessage, but ["+rawMessage.getClass().getName()+"]", e);
-			return null;
-		}
```

Comment:
Beetje vaag dit? Zit er nog een dergelijke check ergens? Of even kijken welk probleem men destijds op probeerde te lossen via Git Blame?

Reply:
de `rawMessage` was vroeger ooit een type `M` en daarvoor `Object`. Dat kon niet altijd worden gecast naar `TextMessage` of zelfs maar `JmsMessage`.

Het vage was dat er hier dus wat code zat die alleen JMS TextMessages accepteerde, maar elders wel ook andere soorten JMS messages kunnen worden verwerkt.

Reply:
> Beetje vaag dit? Zit er nog een dergelijke check ergens? Of even kijken welk probleem men destijds op probeerde te lossen via Git Blame?

Zit sinds 2004 zo in de code toen het `Object rawMessage` was. Dat is de eerste versie die ik kan terug traceren. Daarna is de code diverse keren "as-is" gemigreerd van `JmsListener` naar `Pulling/PushingJmsListener` naar `JmsListenerBase`.